kong/clustering/rpc/json_rpc_v2.lua

```diff
@@ -23,7 +23,10 @@ local ERROR_MSG = {
 
 
 function _M.new_error(id, code, msg)
-  if not msg then
+  if msg then
+    msg = assert(tostring(msg))
```

Comment:
``` suggestion
  if msg then
    local typ = type(msg)
    if typ ~= "string" then
      local mt = getmetatable(msg)
      -- other types without the metamethod `__tostring` don't
      -- generate a meaningful string, we should consider it as a
      -- bug since we should not expose something like
      -- `"table: 0x7fff0000"` to the RPC caller.
      assert(mt.__tostring)
    
```

Reply:
applied